src/ranch_ssl.erl

```diff
@@ -144,6 +148,16 @@ disallowed_listen_options() ->
 		fallback, server_name_indication, srp_identity
 		|ranch_tcp:disallowed_listen_options()].
 
+unsupported_tls_options(SocketOpts) ->
+	unsupported_tls_version_options(get_tls_versions(SocketOpts)).
+
+unsupported_tls_version_options(['tlsv1.3']) ->
```

Comment:
I don't know about that. What happens when the configuration has both TLS 1.2 and 1.3 and those options are given?

Reply:
In that case, it's ok to provide the non-TLSv1.3 options. `ssl:listen/2` will accept it and apply the respective options accordingly when a connection is made.

Reply:
Even when a TLS 1.3 connection is made?

Reply:
I haven't checked that TBH (but will do just to make sure), but `ssl:assert_option_dependency` which is used for the dependency checking builds 2 sets of the requested versions and the allowed versions for that option, and returns `ok` if the intersection of the two sets is not empty, otherwise throws an error. So judging from that, I would say that yes, even if a TLSv1.3 connection is made.

Reply:
... the ranch tests fail, looks like the dependency checking wasn't done before OTP 23 🙄

Reply:
> Even when a TLS 1.3 connection is made?

Confirmed.

Reply:
That's curious but OK.